### PR TITLE
Added REST commands for bot

### PR DESCRIPTION
**Pull request by Steven-Boston**

### CODE REVIEW

**Review by joshhaddock88 — APPROVED** [review]

File: Example/BuddyBot/BuddyBot/DTO/SupportTicketDTO.cs [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+using System.ComponentModel.DataAnnotations;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace BuddyBot.DTO
+{
+    class SupportTicketDTO
+    {
+        public int Id { get; set; }
```
Comment (edy-cf) [inline Example/BuddyBot/BuddyBot/DTO/SupportTicketDTO.cs:12]: Is this required? Please check and add an annotation if needed.

File: Example/BuddyBot/BuddyBot/Modules/Commands.cs [hunk 2]
```diff
@@ -34,71 +39,159 @@ public async Task Moderator()
         {
             GuildPermissions newSet = new GuildPermissions(
                 //bool createInstantInvite =
```
Comment (edy-cf) [inline Example/BuddyBot/BuddyBot/Modules/Commands.cs:41]: Please remove commented-out code

File: Example/BuddyBot/BuddyBot/Modules/Commands.cs [hunk 3]
```diff
@@ -34,71 +39,159 @@ public async Task Moderator()
...
 
             await ReplyAsync("Pong");
         }
 
+        [Command("help")]
+        public async Task Help()
+        {
+            string sender = Context.User.Username;
+
+            await Context.User.SendMessageAsync($"Hello {sender}! Thank you for using SupportBot!\n  To submit a ticket, please use the command !newTicket here followed by a brief description of your troubles. A moderator will assist you shortly.");
```
Comment (edy-cf) [inline Example/BuddyBot/BuddyBot/Modules/Commands.cs:112]: Please don't hardcode text; make this a separate variable

File: Example/BuddyBot/BuddyBot/Modules/Commands.cs [hunk 4]
```diff
@@ -34,71 +39,159 @@ public async Task Moderator()
...
+                Created = DateTime.Now,
+                Description = description,
+                Requestor = sender
+            };
+
+            request.AddJsonBody(newTicket);
+
+            var response = client.Post(request);
+
+            Console.WriteLine(response.StatusCode.ToString() + "      " + response.Content.ToString());
```
Comment (edy-cf) [inline Example/BuddyBot/BuddyBot/Modules/Commands.cs:133]: Please remove console log in final code

File: Example/BuddyBot/BuddyBot/Modules/Commands.cs [hunk 5]
```diff
@@ -34,71 +39,159 @@ public async Task Moderator()
...
+                Resolver = sender,
+                Closed = DateTime.Now
+            };
+            Console.WriteLine("made the DTO");
+
+            request.AddJsonBody(closing);
+
+            Console.WriteLine("added the DTO to the Body");
+
+            //TODO: Figure out why the response is bugged
```
Comment (edy-cf) [inline Example/BuddyBot/BuddyBot/Modules/Commands.cs:186]: Please investigate before merging

File: Example/BuddyBot/BuddyBot/Modules/help.cs [hunk 6]
```diff
@@ -1,12 +1,22 @@
...
 using System.Threading.Tasks;
 
+
 namespace BuddyBot.Modules
 {
-    class help
+    class help : ModuleBase<SocketCommandContext>
     {
+        [Command("help")]
+        public async Task Help()
```
Comment (edy-cf) [inline Example/BuddyBot/BuddyBot/Modules/help.cs:15]: Please rename this HelpAsync() or similar

File: ticketBotApi/ticketBotApi/Migrations/TicketBotDbContextModelSnapshot.cs [hunk 7]
```diff
@@ -0,0 +1,306 @@
...
+using ticketBotApi.Data;
+
+namespace ticketBotApi.Migrations
+{
+    [DbContext(typeof(TicketBotDbContext))]
+    partial class TicketBotDbContextModelSnapshot : ModelSnapshot
+    {
+        protected override void BuildModel(ModelBuilder modelBuilder)
+        {
+#pragma warning disable 612, 618
```
Comment (edy-cf) [inline ticketBotApi/ticketBotApi/Migrations/TicketBotDbContextModelSnapshot.cs:16]: Please comment on why this is needed

File: ticketBotApi/ticketBotApi/bin/Debug/net5.0/ticketBotApi.runtimeconfig.dev.json [hunk 8]
```diff
@@ -1,8 +1,8 @@
 {
   "runtimeOptions": {
     "additionalProbingPaths": [
-      "C:\\Users\\joshh\\.dotnet\\store\\|arch|\\|tfm|",
-      "C:\\Users\\joshh\\.nuget\\packages"
+      "C:\\Users\\steve\\.dotnet\\store\\|arch|\\|tfm|",
```
Comment (edy-cf) [inline ticketBotApi/ticketBotApi/bin/Debug/net5.0/ticketBotApi.runtimeconfig.dev.json:4]: Please change this to dummy value

**Review by edy-cf — COMMENTED** [review]
Code review completed

**Pull request merged**